### PR TITLE
email field model convert

### PR DESCRIPTION
**Pull request by psantos1113**
With the old rest APIs convert field models into their static configuration.

### CODE REVIEW

**Review by ChomickiM — APPROVED** [review]

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+@Component
+public class EmailGlobalFieldModelConverter implements GlobalFieldModelToConcreteConverter<EmailGlobalConfigModel> {
+    private Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String EMAIL_FIELD_PREFIX = "mail.smtp.";
+    public static final String AUTH_REQUIRED_KEY = EMAIL_FIELD_PREFIX + "auth";
+    public static final String EMAIL_FROM_KEY = EMAIL_FIELD_PREFIX + "from";
+    public static final String EMAIL_HOST_KEY = EMAIL_FIELD_PREFIX + "host";
+    public static final String AUTH_PASSWORD_KEY = EMAIL_FIELD_PREFIX + "password";
+    public static final String EMAIL_PORT_KEY = EMAIL_FIELD_PREFIX + "port";
+    public static final String AUTH_USER_KEY = EMAIL_FIELD_PREFIX + "user";
```
Comment (bamandel) [inline channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java:35]: Shouldn't these keys be available in EmailPropertyKeys? Is that not reachable from this class?

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            globalFieldModel.getFieldValue(EMAIL_PORT_KEY)
+                .map(Integer::valueOf)
+                .ifPresent(model::setSmtpPort);
+            globalFieldModel.getFieldValue(AUTH_REQUIRED_KEY)
+                .map(Boolean::valueOf)
+                .ifPresent(model::setSmtpAuth);
+            globalFieldModel.getFieldValue(AUTH_USER_KEY)
+                .ifPresent(model::setSmtpUsername);
+            globalFieldModel.getFieldValueModel(AUTH_PASSWORD_KEY)
+                .filter(Predicate.not(FieldValueModel::getIsSet))
```
Comment (bamandel) [inline channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java:69]: Do we want to check isSet here? If the user passes a value in the password section, I feel that we should set that value regardless of whether or not the isSet field is true or false.

Comment (bamandel) [reply channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java:69]: I guess we do need to check if it's set in the case that it's empty we need to insert the original value, right?

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java (hunk 1 above)
Comment (psantos1113) [reply channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java:35]: yes we should be able to use this instead.

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java (hunk 2 above)
Comment (psantos1113) [reply channel-email/src/main/java/com/synopsys/integration/alert/channel/email/convert/EmailGlobalFieldModelConverter.java:69]: I'm going to look into this.

**Comment by psantos1113** [comment]
Found an issue while performing additional testing closing this PR.

**Pull request closed without merging**